uaclient/tests/test_config.py

```diff
@@ -0,0 +1,172 @@
+import json
+import os
+
+from nose2.tools.params import params
+
+from uaclient.config import UAConfig
+from uaclient.testing.helpers import TestCase
+
+
+# These are in a variable rather than inline to work around
+# https://github.com/nose-devs/nose2/issues/433
+SIMPLE_PARAMS = (('machine_token', 'machine-token', None),
+                 ('contracts', 'account-contracts', []))
+
+KNOWN_DATA_PATHS = (('bound-macaroon', 'bound-macaroon'),
+                    ('accounts', 'accounts.json'))
+
+
+class TestAccounts(TestCase):
+
+    def test_accounts_returns_empty_list_when_no_cached_account_value(self):
+        """Config.accounts property returns an empty list when no cache."""
+        tmp_dir = self.tmp_dir()
+        cfg = UAConfig({'data_dir': tmp_dir})
+
+        assert [] == cfg.accounts
+
+    def test_accounts_extracts_accounts_key_from_read_cache(self):
+        """Config.accounts property extracts the accounts key from cache."""
+        tmp_dir = self.tmp_dir()
+        cfg = UAConfig({'data_dir': tmp_dir})
+        cfg.write_cache('accounts', {'accounts': ['acct1', 'acct2']})
+
+        assert ['acct1', 'acct2'] == cfg.accounts
+
+
+class TestDataPath(TestCase):
+
+    def test_data_path_returns_data_dir_path_without_key(self):
+        """The data_path method returns the data_dir when key is absent."""
+        cfg = UAConfig({'data_dir': '/my/dir'})
```

Comment:
We should consider using a temporary directory here (and below); it wouldn't be surprising for `UAConfig` to ensure its `data_dir` exists at instantiation (rather than on first write as is currently the case), and that could cause some interesting problems.

Reply:
"it wouldn't be surprising for UAConfig to ensure its data_dir exists at instantiation".  

I guess I don't have a strong case here, but I guess from the Landscape projects we had tried avoiding side-effects in __init__  just to avoid work on instantiation and testing related to that init. I *think* that suggestion really comes from old C++ constructor concerns and python style choices more than actual defensible python programming practices. So, I'm game for moving it into __init__. Though I'd generally like to avoid complexity in __init__ if we can. (This case doesn't really suggest complexity).

Reply:
Yep, I broadly agree that we shouldn't do it; this was me overthinking how robust our tests need to be.